### ISSUE TITLE
Cannot update constraint

### ISSUE THREAD

**Issue by phamquochoan** [SnapKit/SnapKit#332]
### New Issue Checklist

🚫 If this template is not filled out your issue **will** be closed with no comment. 🚫

* [x] I have looked at the [Documentation](http://snapkit.io/docs)
* [x] I have read the [F.A.Q.](http://snapkit.io/faq)
* [x] I have filled out this issue template.

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform                | iOS (Xcode 8, swift 3)
 Platform Version        | 8.4
 SnapKit Version         | 3.0.1
 Integration Method      |  carthage
 

### Issue Description

I've migrated my projects to Swift 3.0, they work fine on iOS 9.x and 10.x
However, this problem only appears on iOS 8.x (reported on simulator and iOS device running iOS 8.4):

```
2016-09-23 10:23:47.828 SnapDemo[13609:131065] *** Assertion failure in -[SnapKit.LayoutConstraint setPriority:], /SourceCache/Foundation_Sim/Foundation-1144.17/Layout.subproj/NSLayoutConstraint.m:174
2016-09-23 10:23:47.831 SnapDemo[13609:131065] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Mutating a priority from required to not on an installed constraint (or vice-versa) is not supported.  You passed priority 1000 and the existing priority was 1000.'
*** First throw call stack:
(
	0   CoreFoundation                      0x00a31746 __exceptionPreprocess + 182
	1   libobjc.A.dylib                     0x006baa97 objc_exception_throw + 44
	2   CoreFoundation                      0x00a315da +[NSException raise:format:arguments:] + 138
	3   Foundation                          0x00327720 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 118
	4   Foundation                          0x002944ef -[NSLayoutConstraint setPriority:] + 397
	5   SnapKit                             0x001f8d88 _TFC7SnapKit10Constraint33updateConstantAndPriorityIfNeededfT_T_ + 1048
	6   SnapKit                             0x001f5b90 _TFC7SnapKit10ConstraintWP33_F1BDC77CAFF662929BB62E80B81AE0B78constantPS_24ConstraintConstantTarget_ + 32
	7   SnapKit                             0x001f5ce5 _TFC7SnapKit10ConstraintsP33_F1BDC77CAFF662929BB62E80B81AE0B78constantPS_24ConstraintConstantTarget_ + 229
	8   SnapKit                             0x001f822e _TFC7SnapKit10Constraint6updatefT6offsetPS_22ConstraintOffsetTarget__S0_ + 158
	9   SnapDemo                            0x000d7dc9 _TFFC8SnapDemo14ViewController11viewDidLoadFT_T_U_FT_T_ + 249
	10  SnapDemo                            0x000d7f68 _TTRXFo___XFdCb___ + 40
	11  libdispatch.dylib                   0x035a75ea _dispatch_call_block_and_release + 15
	12  libdispatch.dylib                   0x035c9bef _dispatch_client_callout + 14
	13  libdispatch.dylib                   0x035af6bb _dispatch_main_queue_callback_4CF + 993
	14  CoreFoundation                      0x0098a8ee __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 14
	15  CoreFoundation                      0x009485f0 __CFRunLoopRun + 2256
	16  CoreFoundation                      0x00947a5b CFRunLoopRunSpecific + 443
	17  CoreFoundation                      0x0094788b CFRunLoopRunInMode + 123
	18  GraphicsServices                    0x044d52c9 GSEventRunModal + 192
	19  GraphicsServices                    0x044d5106 GSEventRun + 104
	20  UIKit                               0x00df00b6 UIApplicationMain + 1526
	21  SnapDemo                            0x000d9f01 main + 145
	22  libdyld.dylib                       0x035f4ac9 start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

Demo code (straight from SnapKit 3.0.1 source code):

```swift
import UIKit
import SnapKit

class ViewController: UIViewController {

    var v1: UIView!
    var v2: UIView!
    
    var c1: Constraint! = nil
    var c2: Constraint! = nil
    
    var didSetupConstraints = false
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        v1 = UIView()
        v1.backgroundColor = UIColor.red.withAlphaComponent(0.5)
        
        v2 = UIView()
        v2.backgroundColor = UIColor.blue.withAlphaComponent(0.5)
        
        view.addSubview(v1)
        view.addSubview(v2)
        
        view.setNeedsUpdateConstraints()
   
        DispatchQueue.main.asyncAfter(deadline: DispatchTime(uptimeNanoseconds: 1000)) { [unowned self] in
            self.c1.update(offset: 15)
            self.c2.update(offset: 20)
        }
        
        
        DispatchQueue.main.asyncAfter(deadline: DispatchTime(uptimeNanoseconds: 2000)) { [unowned self] in
            self.c1.update(inset: 15)
            self.c2.update(inset: 20)
        }
        
    }
    
    override func updateViewConstraints() {
        guard !didSetupConstraints else  { return }
        
        v1.snp.makeConstraints { (make) -> Void in
            c1 = make.top.equalTo(v2).offset(50).constraint
            c2 = make.bottom.equalTo(v2).offset(25).constraint
            return
        }
        
        didSetupConstraints = true
    
        super.updateViewConstraints()
    }

}
```

**Comment by kimsungwhee** [comment]
same problem

**Comment by robertjpayne** [comment]
This is related to #323 which will be pushed out next week, you can target develop branch if you need the fix earlier.

**Comment by studioofthemonth** [comment]
@phamquochoan does targeting the develop branch 3.0.1 fix this problem for you? I'm still getting a crash when calling update(offset:) but I haven't dug into my problem enough to know for sure it's the same thing.

**Comment by phamquochoan** [comment]
@studioofthemonth I got no problem after switching to develop branch. It maybe a different error I guess.

**Comment by pixeldock** [comment]
I can confirm that using the development branch fixes the crash.

**Comment by Fogh** [comment]
I had this crash as well and I can also confirm that this is fixed on the development branch.

**Comment by robertjpayne** [comment]
Fixed with [3.0.2](https://github.com/SnapKit/SnapKit/releases/tag/3.0.2)
